File: LeetCode/242.valid-anagram.java
/*
 * @lc app=leetcode id=242 lang=java
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {

        int n1 = s.size();
        int n2 = t.size();

        Map<Character, Integer> mp = new HashMap<>();

        for (var it : s) {
            mp[it]++;
        }

        for (var it : t) {
            mp[it]--;
        }

        for (var it : mp) {
            if (it.second != 0) {
                return false;
            }
        }

        return true;

    }
}
// @lc code=end

File: LeetCode/242.valid-anagram.py
#
# @lc app=leetcode id=242 lang=python3
#
# [242] Valid Anagram
#

# @lc code=start
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        # 1. Sorting
        return sorted(s) == sorted(t)

        # # 2. Hash Table
        # if len(s) != len(t):
        #     return False
        # count = {}
        # for c in s:
        #     if c in count:
        #         count[c] += 1
        #     else:
        #         count[c] = 1
        # for c in t:
        #     if c in count:
        #         count[c] -= 1
        #     else:
        #         return False
        # for c in count:
        #     if count[c] != 0:
        #         return False
        # return True



       
        
# @lc code=end

